var/spack/repos/builtin/packages/py-petsc4py/package.py

```diff
@@ -62,6 +63,7 @@ class PyPetsc4py(PythonPackage):
     depends_on("petsc+mpi", when="+mpi")
     depends_on("petsc~mpi", when="~mpi")
     depends_on("petsc@main", when="@main")
+    depends_on("petsc@3.19.0:3.19", when="@3.19.0:3.19")
```

Comment:
If you want to simplify these, you could do:
```python
depends_on("petsc@3.19", when="@3.19")
```
You could even put this in a for-loop:
```python
for ver in ["3.19", ..., "3.11"]:
    depends_on(f"petsc@{ver}", when=f"@{ver}")
```

Reply:
Thanks! I there are other places also where it can use cleanup. Will do this in a subsequent PR.

For now - I think this one is ready for merge.